code/game/jobs/job/engineering.dm

```diff
@@ -43,6 +43,40 @@
 	dufflebag = /obj/item/weapon/storage/backpack/duffel/engineering
 	box = /obj/item/weapon/storage/box/engineer
 
+/datum/job/senioreng

```

Comment:
Most of these variables are the same, right? Why not just make all of these senior roles a subtype of the job itself? As long as they have unique flags, I can't think of any reason that could go wrong.